src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -0,0 +1,956 @@
+                            StandardWatchEventKinds.ENTRY_DELETE, StandardWatchEventKinds.ENTRY_CREATE },
+                    SensitivityWatchEventModifier.HIGH);
+            debug("Watching build file directory: " + buildFile.getParentFile().toPath());
+
+            while (true) {
+                // check if javaSourceDirectory has been added
+                if (!sourceDirRegistered && this.sourceDirectory.exists()) {
+                    compile(this.sourceDirectory);
+                    registerAll(this.sourceDirectory.toPath(), srcPath, watcher);
+                    debug("Registering Java source directory: " + this.sourceDirectory);
+                    sourceDirRegistered = true;
+                } else if (sourceDirRegistered && !this.sourceDirectory.exists()) {
+                    cleanTargetDir(outputDirectory);
+                    sourceDirRegistered = false;
+                }
+
+                // check if testSourceDirectory has been added
+                if (!testSourceDirRegistered && this.testSourceDirectory.exists()) {
+                    compile(this.testSourceDirectory);
+                    registerAll(this.testSourceDirectory.toPath(), testSrcPath, watcher);
+                    debug("Registering Java test directory: " + this.testSourceDirectory);
+                    runTestThread(false, executor, -1, false, false);
+                    testSourceDirRegistered = true;
+                } else if (testSourceDirRegistered && !this.testSourceDirectory.exists()) {
+                    cleanTargetDir(testOutputDirectory);
+                    testSourceDirRegistered = false;
+                }
+                
+                // check if configDirectory has been added
+                if (!configDirRegistered && this.configDirectory.exists()){
+                    configDirRegistered = true;
+                    debug("Configuration directory has been added: " + this.configDirectory);
+                    info("The server configuration directory " + configDirectory + " has been added. Restart liberty:dev mode for it to take effect.");
+                }
+                
+                // check if configFile has been added
+                if (!configFileRegistered && configFile.exists()){
+                    configFileRegistered = true;
+                    debug("Configuration file has been added: " + configFile);
+                    info("The server configuration file " + configFile + " has been added. Restart liberty:dev mode for it to take effect.");
+                }
+                
+                try {
+                    final WatchKey wk = watcher.poll(1, TimeUnit.SECONDS);
+                    for (WatchEvent<?> event : wk.pollEvents()) {
+                        final Path changed = (Path) event.context();
+
+                        final Watchable watchable = wk.watchable();
+                        final Path directory = (Path) watchable;
```

Comment:
Is there any concern that will not be the canonical path?

Reply:
This `java.nio.file.Path` object used here will be a relative path to the watched directory `typically represent a system dependent file path`.  It does not need to be the canonical path in this case.  We then calculate the fileChanged path on line 481: 
```java
File fileChanged = new File(directory.toString(), changed.toString());
```